src/ModelContextProtocol/Server/McpServerToolAttribute.cs

```diff
@@ -1,18 +1,70 @@
 ﻿namespace ModelContextProtocol.Server;
 
 /// <summary>
-/// Used to mark a public method as an MCP tool.
+/// Used to indicate that a method should be considered an MCP tool and describe it.
 /// </summary>
 [AttributeUsage(AttributeTargets.Method)]
 public sealed class McpServerToolAttribute : Attribute
 {
+    private bool? _destructive;
+    private bool? _idempotent;
+    private bool? _openWorld;
+    private bool? _readOnly;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="McpServerToolTypeAttribute"/> class.
+    /// </summary>
+    public McpServerToolAttribute()
+    {
+    }
+
     /// <summary>Gets the name of the tool.</summary>
     /// <remarks>If <see langword="null"/>, the method name will be used.</remarks>
-    public string? Name { get; }
+    public string? Name { get; set; }
 
     /// <summary>
-    /// Initializes a new instance of the <see cref="McpServerToolTypeAttribute"/> class.
+    /// Gets or sets a human-readable title for the tool.
+    /// </summary>
+    public string? Title { get; set; }
+
+    /// <summary>
+    /// Gets or sets whether the tool may perform destructive updates to its environment.
+    /// </summary>
+    public bool Destructive { get => _destructive ?? true; set => _destructive = value; }
+
+    /// <summary>
+    /// Gets whether the <see cref="Destructive"/> property has been assigned a value.
+    /// </summary>
+    public bool DestructiveIsSet => _destructive.HasValue;
+
+    /// <summary>
+    /// Gets or sets whether calling the tool repeatedly with the same arguments will have no additional effect on its environment.
+    /// </summary>
+    public bool Idempotent { get => _idempotent ?? false; set => _idempotent = value; }
+
+    /// <summary>
+    /// Gets whether the <see cref="Idempotent"/> property has been assigned a value.
+    /// </summary>
+    public bool IdempotentIsSet => _idempotent.HasValue;
```

Comment:
I take it you'll be marking this and other similar properties internal for now?

Reply:
Yes. I've removed them and made the backing fields internal for now.